networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -7,23 +7,44 @@
  */
 
 #include <atomic>
-#include <cstring>
 #include <fstream>
 
 #include <networkit/auxiliary/Log.hpp>
 #include <networkit/io/NetworkitBinaryReader.hpp>
 #include <networkit/io/NetworkitBinaryGraph.hpp>
 #include <networkit/io/MemoryMappedFile.hpp>
 
+template <class T>
+const char *getIterator(const T &source);
+
+template <>
+const char *getIterator(const NetworKit::MemoryMappedFile &source) {
+    return source.cbegin();
+}
+
+template <>
+const char *getIterator(const std::string &source) {
+    return source.data();
+}
```

Comment:
Do not use specialization here. Use multiple function overloads instead (basically, just drop the `template <>`).

Reply:
These functions are local to the file and should be in an anonymous namespace.

Reply:
Since these functions return a `const char *`, `accessData()` would be a better name.